beacon-chain/sync/rpc_status.go

```diff
@@ -36,7 +38,14 @@ func (r *RegularSync) maintainPeerStatuses() {
 					}
 				}
 			}
-
+			_, highestEpoch := r.p2p.Peers().HighestFinalizedPeer()
+			if highestEpoch > r.chain.FinalizedCheckpt().Epoch {
+				r.clearPendingSlots()
+				// block until we can resync the node
```

Comment:
What does this comment mean?

Reply:
since this is called in the `maintainPeerStatuses` loop , we wait for the resync to finish completely instead of launching it into a separate routine.